R/pkratio-datamapping.R

```diff
@@ -9,30 +9,86 @@ PKRatioDataMapping <- R6::R6Class(
     colorGrouping = NULL,
     sizeGrouping = NULL,
     shapeGrouping = NULL,
-
+    
+    colorGroupingDataFrame = NULL,
+    sizeGroupingDataFrame = NULL,
+    shapeGroupingDataFrame = NULL,
+    
+    colorLegendTitle = NULL,
+    sizeLegendTitle = NULL,
+    shapeLegendTitle = NULL,
+    
     # Example of how to do some other stuff
     initialize = function(x = "Age",
                           y = "Ratio",
                           colorGrouping = NULL,
                           sizeGrouping = NULL,
-                          shapeGrouping = NULL) {
-      self$colorGrouping <- colorGrouping
-      self$sizeGrouping <- sizeGrouping
-      self$shapeGrouping <- shapeGrouping
-
+                          shapeGrouping = NULL,
+                          colorGroupingDataFrame = NULL,
+                          sizeGroupingDataFrame = NULL,
+                          shapeGroupingDataFrame = NULL,
+                          colorLegendTitle = NULL,
+                          sizeLegendTitle = NULL,
+                          shapeLegendTitle = NULL) {
+      
+      colorGroupingList <- self$get_grouping_list(colorGrouping,colorGroupingDataFrame)
+      sizeGroupingList  <- self$get_grouping_list(sizeGrouping,sizeGroupingDataFrame)
+      shapeGroupingList <- self$get_grouping_list(shapeGrouping,shapeGroupingDataFrame)
+      
+      self$colorGrouping <- colorGroupingList$selfGrouping
+      self$sizeGrouping  <- sizeGroupingList$selfGrouping
+      self$shapeGrouping <- shapeGroupingList$selfGrouping
+      
+      self$colorGroupingDataFrame <- colorGroupingList$selfGroupingDataFrame
+      self$sizeGroupingDataFrame  <- sizeGroupingList$selfGroupingDataFrame
+      self$shapeGroupingDataFrame <- shapeGroupingList$selfGroupingDataFrame
+      
+      self$colorLegendTitle = colorLegendTitle
+      self$sizeLegendTitle = sizeLegendTitle
+      self$shapeLegendTitle = shapeLegendTitle
+      
       super$initialize(x, y)
     },
-
+    
+    
+    get_grouping_list = function(Grouping,GroupingDataFrame){  #move out of public?

```

Comment:
ok here our coding standard is a bit off
 * Methods should be without `_` => getGroupingList
 * Lowercase for parameters (grouping, groupingDataFrame) etc..